src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1618,6 +1659,11 @@ public int ReceiveFrom(byte[] buffer, int offset, int size, SocketFlags socketFl
 
             int bytesTransferred;
             SocketError errorCode = SocketPal.ReceiveFrom(_handle, buffer, offset, size, socketFlags, socketAddress.Buffer, ref socketAddress.InternalSize, out bytesTransferred);
+            if (SocketsTelemetry.Log.IsEnabled())
+            {
+                SocketsTelemetry.Log.BytesReceived(bytesTransferred);
+                if (SocketType == SocketType.Dgram) SocketsTelemetry.Log.DatagramReceived();
```

Comment:
If I'm sent a datagram containing 10 bytes, and I make 10 calls to ReceiveFrom each to read 1 byte, what happens / how many DatagramReceived events are logged?

Reply:
According to [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.receivefrom?view=netcore-3.1#System_Net_Sockets_Socket_ReceiveFrom_System_Byte___System_Net_EndPoint__), datagram cannot be read partially and a call to `ReceiveFrom` either returns a complete datagram or blocks or throw an exception.

>With connectionless protocols, ReceiveFrom will read the first enqueued datagram received into the local network buffer. If the datagram you receive is larger than the size of buffer, the ReceiveFrom method will fill buffer with as much of the message as is possible, and throw a SocketException. If you are using an unreliable protocol, the excess data will be lost. If you are using a reliable protocol, the excess data will be retained by the service provider and you can retrieve it by calling the ReceiveFrom method with a large enough buffer.

Reply:
Ok.